app/uk/gov/hmrc/hec/controllers/TaxCheckController.scala

```diff
@@ -108,4 +109,26 @@ class TaxCheckController @Inject() (
       )
   }
 
+  val saveEmailAddress: Action[JsValue] = authenticateGG.async(parse.json) { implicit request =>
+    Json.fromJson[SaveEmailAddressRequest](request.body) match {
+      case JsSuccess(saveEmailAddressRequest, _) =>
+        taxCheckService
+          .saveEmailAddress(saveEmailAddressRequest)
+          .fold(
+            { e =>
```

Comment:
we can use _.doThrow method here? I am not sure if it was created here or in FE

Reply:
that was created in the FE to make sure we were displaying the correct technical difficulties page to the user - without it it was showing a blank screen with the exception message on it I think. There's not that concern in the BE so we should be able to get away without it I think